jgrapht-core/src/main/java/org/jgrapht/traverse/RandomWalkIterator.java

```diff
@@ -17,247 +17,150 @@
  */
 package org.jgrapht.traverse;
 
-import org.jgrapht.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.Collectors;
 
-import java.util.*;
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
 
 /**
- * A random walk iterator for a directed or undirected graph.
+ * A random walk iterator.
  * 
- * <p>
- * At each step the iterator selects a random (uniformly distributed) edge out of the current vertex
- * and follows it to the next vertex. In case of directed graphs the outgoing edge set is used. See
- * <a href="https://en.wikipedia.org/wiki/Random_walk#Random_walk_on_graphs">wikipedia</a> for more
- * details.
+ * "Given a graph and a starting point, we select a neighbor of it at random, and move to this
+ * neighbor; then we select a neighbor of this point at random, and move to it etc. The (random)
+ * sequence of points selected this way is a random walk on the graph." This very simple definition,
+ * together with a comprehensive survey can be found at: "Lovász, L. (1993). Random walks on graphs:
+ * A survey. Combinatorics, Paul erdos is eighty, 2(1), 1-46."
  * 
- * <p>
- * In case a weighted walk is desired, edges are selected with probability respective to its weight
- * (out of the total weight of the edges). The walk can be bounded by number of steps (default
- * {@code Long#MAX_VALUE} . When the bound is reached the iterator is considered exhausted. Calling
- * {@code next()} on exhausted iterator will throw {@code NoSuchElementException}.
+ * In its default variant the probability of selecting an outgoing edge is one over the (out) degree
+ * of the vertex. In case the user requests a weighted walk, then the probability of each edge is
+ * equal to its weight divided by the total weight of all outgoing edges. The walk can also be
+ * bounded by a maximum number of hops (edges traversed). The iterator returns
+ * {@link NoSuchElementException} when this bound is reached.
  * 
- * In case a sink (i.e. no edges) vertex is reached, any consecutive calls to {@code next()} will
- * throw {@code NoSuchElementException}.
- * 
- * <p>
- * For this iterator to work correctly the graph must not be modified during iteration. Currently
- * there are no means to ensure that, nor to fail-fast. The results of such modifications are
- * undefined.
- * 
- * @author Assaf Mizrachi
+ * @author Dimitrios Michail
  *
- * @param <V> vertex type
- * @param <E> edge type
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
  */
 public class RandomWalkIterator<V, E>
-    extends
-    AbstractGraphIterator<V, E>
```

Comment:
Although I said I was OK with making the behavioral change directly (whether to return the start vertex), I don't think we should make actual breaking changes to the interface itself.  Meaning either keep it as AbstractGraphIterator and deprecate the no-vertex constructor, or deprecate the entire class and create a new one.  If we go with the latter, then suggested name for new class is RandomWalkVertexIterator.